porch/pkg/cache/repository.go

```diff
@@ -373,13 +398,48 @@ func (r *cachedRepository) refreshAllCachedPackages(ctx context.Context) (map[re
 			klog.Warningf("over-notifying of package updates (even on unchanged packages)")
 			r.objectCache.notifyPackageRevisionChange(watch.Modified, newPackage)
 		}
+		// If a new package is found that doesn't have a corresponding PackageRevisionMeta,
+		// we create one.
+		if _, found := existingPkgRevMetasMap[newPackage.KubeObjectName()]; !found {
+			pkgRevMeta := meta.PackageRevisionMeta{
+				Name:      newPackage.KubeObjectName(),
+				Namespace: newPackage.KubeObjectNamespace(),
+			}
+			if _, err := r.metadataStore.Create(ctx, pkgRevMeta, r.repoSpec); err != nil {
+				return nil, nil, err
```

Comment:
hmm...shouldn't we continue to refresh other packages ?

Reply:
I've updated the logic here to make it simpler and also handle automatic retries of creation of the `PackageRev` CRD on every sync.